common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/ObjectTracker.java

```diff
@@ -165,23 +165,50 @@ private boolean checkAndSpeakIfAllGroupsEmpty() {
     }
 
     private void targetNearestObject() {
-        List<Entity> entities = POIEntities.getInstance().getLastScanResults();
-        List<BlockPos> blocks = POIBlocks.getInstance().getLastScanResults();
-
-        if (!entities.isEmpty() && blocks.isEmpty()) currentObject = entities.getFirst();
-        if (!blocks.isEmpty() && entities.isEmpty()) currentObject = blocks.getFirst();
-        if (!entities.isEmpty() && !blocks.isEmpty()) {
-            LocalPlayer player = WorldUtils.getClientPlayer();
+        LocalPlayer player = WorldUtils.getClientPlayer();
+        List<Entity> entities = POIEntities.getInstance().getLastScanResults()
+        .stream().sorted((a, b) -> (int)(a.distanceTo(player) - b.distanceTo(player)))
+        .toList();
+
+        List<BlockPos> blocks = POIBlocks.getInstance().getLastScanResults()
+        .stream().sorted((a, b) -> (int)(player.getEyePosition().distanceTo(a.getCenter()) - player.getEyePosition().distanceTo(b.getCenter())))
+        .toList();
+
+        /*
+         * 0 = not found
+         * 1 = entity
+         * 2 = block
+         * 3 = both
+        */
+        int status = 0;
+
+        if (!entities.isEmpty() && (blocks.isEmpty() || Screen.hasControlDown())) {
+            currentObject = entities.getFirst();
+            status = Screen.hasControlDown() ? 1 : 3;
+        } else if (entities.isEmpty() && Screen.hasControlDown()) status = -1;
+        else if (!blocks.isEmpty() && (entities.isEmpty() || Screen.hasShiftDown())) {
+            currentObject = blocks.getFirst();
+            status = Screen.hasShiftDown() ? 2 : 3;
+        } else if (blocks.isEmpty() && Screen.hasShiftDown()) status = -2;
+        else if (!entities.isEmpty() && !blocks.isEmpty()) {
```

Comment:
These else-ifs should all be braced for consistency.
```suggestion
        if (!entities.isEmpty() && (blocks.isEmpty() || Screen.hasControlDown())) {
            currentObject = entities.getFirst();
            status = Screen.hasControlDown() ? 1 : 3;
        } else if (entities.isEmpty() && Screen.hasControlDown()) {
            status = -1;
        } else if (!blocks.isEmpty() && (entities.isEmpty() || Screen.hasShiftDown())) {
            currentObject = blocks.getFirst();
            status = Screen.hasShiftDown() ? 2 : 3;
        } else if (blocks.isEmpty() && Screen.hasShiftDown()) {
            status = -2;
        } else if (!entities.isEmpty() && !blocks.isEmpty()) {
```